### PR TITLE
feature/private routing

### PR DESCRIPTION
**Pull request by PatrickConnollyONS**
### What

Add a private-route-prefix, which, when used with the frontend-router, allows for a private route for the release calendar controller

### How to review

Review of code and ensure tests are passing

### Who can review

Anyone

### CODE REVIEW

File: mapper/mapper.go [hunk 1]
```diff
@@ -441,9 +441,14 @@ func getWindowOffset(windowSize int) int {
...
 
-func calendarEntryFromRelease(release search.Release) model.CalendarEntry {
+func calendarEntryFromRelease(release search.Release, uriPrivatePrefix string) model.CalendarEntry {
 	result := model.CalendarEntry{
-		URI:         release.URI,
+		URI: func(uri, privateSuffix string) string {
+			if privateSuffix == "" {
+				return release.URI
+			}
+			return privateSuffix + release.URI
```
Comment (s-xton) [inline mapper/mapper.go:450]: should we ensure that there is a slash here, i.e. release.URI always starts with /

File: handlers/handlers.go [hunk 2]
```diff
@@ -36,10 +37,11 @@ func Release(cfg config.Config, rc RenderClient, api ReleaseCalendarAPI) http.Ha
 	})
 }
 
-func release(w http.ResponseWriter, req *http.Request, userAccessToken, collectionID, lang string, rc RenderClient, api ReleaseCalendarAPI, _ config.Config) {
+func release(w http.ResponseWriter, req *http.Request, userAccessToken, collectionID, lang string, rc RenderClient, api ReleaseCalendarAPI, cfg config.Config) {
 	ctx := req.Context()
 
-	release, err := api.GetLegacyRelease(ctx, userAccessToken, collectionID, lang, req.URL.EscapedPath())
+	_, uri, _ := strings.Cut(req.URL.EscapedPath(), cfg.PrivateRoutingPrefix)
```
Comment (s-xton) [inline handlers/handlers.go:43]: trim prefix, in place of cut ?

**Review by s-xton — CHANGES_REQUESTED** [review]

File: mapper/mapper.go (hunk 1 above)
Comment (PatrickConnollyONS) [reply mapper/mapper.go:450]: yes I think that's a good idea. I always assumed we would have a path starting with a '/', but no harm checking and adding one if it's not there

File: handlers/handlers.go (hunk 2 above)
Comment (PatrickConnollyONS) [reply handlers/handlers.go:43]: no I explicitly want to remove the prefix, so I have the correct path for retrieval from zebedee/relcal-api

Comment (PatrickConnollyONS) [reply handlers/handlers.go:43]: I beg your pardon :-) Yes indeed strings.TrimPrefix() is perfectly fine and better suited

File: mapper/mapper.go (hunk 1 above)
Comment (PatrickConnollyONS) [reply mapper/mapper.go:450]: Hi Steve added the check into the config validation (only done once then but may be overkill?), have a look

**Pull request closed without merging**